tests/integration/test_storage_s3_queue/test.py

```diff
@@ -847,107 +848,87 @@ def test_max_set_age(started_cluster):
+        node.query(f"SELECT uniq(_path) from {dst_table_name}")
+    )
 
     paths_count = [
         int(x)
         for x in node.query(
             f"SELECT count() from {dst_table_name} GROUP BY _path"
         ).splitlines()
     ]
-    assert 10 == len(paths_count)
+    assert files_to_generate == len(paths_count)
     for path_count in paths_count:
         assert 2 == path_count
 
-    failed_count = int(
-        node.query(
-            "SELECT value FROM system.events WHERE name = 'ObjectStorageQueueFailedFiles' SETTINGS system_events_show_zero_values=1"
+    def get_object_storage_failures():
+        return int(
+            node.query(
+                "SELECT value FROM system.events WHERE name = 'ObjectStorageQueueFailedFiles' SETTINGS system_events_show_zero_values=1"
+            )
         )
-    )
+
+    failed_count = get_object_storage_failures()
 
     values = [
         ["failed", 1, 1],
     ]
     values_csv = (
         "\n".join((",".join(map(str, row)) for row in values)) + "\n"
     ).encode()
-    put_s3_file_content(started_cluster, f"{files_path}/fff.csv", values_csv)
 
-    for _ in range(30):
-        if failed_count + 1 == int(
-            node.query(
-                "SELECT value FROM system.events WHERE name = 'ObjectStorageQueueFailedFiles' SETTINGS system_events_show_zero_values=1"
-            )
-        ):
-            break
-        time.sleep(1)
+    # use a different filename for each test to allow running a bunch of them sequentially with --count
+    file_with_error = f"max_set_age_fail_{uuid4().hex[:8]}.csv"
+    put_s3_file_content(started_cluster, f"{files_path}/{file_with_error}", values_csv)
 
-    assert failed_count + 1 == int(
-        node.query(
-            "SELECT value FROM system.events WHERE name = 'ObjectStorageQueueFailedFiles' SETTINGS system_events_show_zero_values=1"
-        )
-    )
+    wait_for_condition(lambda: failed_count + 1 <= get_object_storage_failures())
 
     node.query("SYSTEM FLUSH LOGS")
     assert "Cannot parse input" in node.query(
-        "SELECT exception FROM system.s3queue WHERE file_name ilike '%fff.csv'"
-    )
-    assert 1 == int(
-        node.query(
-            "SELECT count() FROM system.s3queue_log WHERE file_name ilike '%fff.csv'"
-        )
+        f"SELECT exception FROM system.s3queue WHERE file_name ilike '%{file_with_error}'"
     )
+
     assert 1 == int(
         node.query(
-            "SELECT count() FROM system.s3queue_log WHERE file_name ilike '%fff.csv' AND notEmpty(exception)"
+            f"SELECT count() FROM system.s3queue_log WHERE file_name ilike '%{file_with_error}' AND notEmpty(exception)"
         )
     )
 
-    time.sleep(max_age + 1)
-
-    assert failed_count + 2 == int(
-        node.query(
-            "SELECT value FROM system.events WHERE name = 'ObjectStorageQueueFailedFiles'"
-        )
-    )
+    wait_for_condition(lambda: failed_count + 2 <= get_object_storage_failures())
```

Comment:
The point here, as I remember, was to check that in `max_set_age` time we increase failed_count only by 1, but now it does not check this (as it accepts more than 1).

Reply:
ah, you explained it in PR description...

> Other tests running in parallel may have any object storage failure, making the equality with exactly +1 or +2 not to be true anymore.

but I thought it is impossible that other tests can be run in parallel with the same clickhouse server. AFAIK tests run in parallel use different clickhouse servers.

Reply:
Yep, that's an unfortunate side-effect. In my case it was convenient to change it to be able to run several of this test in parallel through `pytest-xsdist` without affecting each other. But then I thought that on the CI you cannot ensure no other test contributes to this error unless we completely isolate the ClickHouse DB for this one. Otherwise, how can we ensure that `system.events` is never going to contain any `ObjectStorageQueueFailedFiles` created by any other test?

AFAIU how the CI works, it runs 5 of them in parallel. We know for sure all tests in `test_storage_s3_queue` never run in parallel. But what about the others? There are other 4 tests running on the same ClickHouse DB that could potentially affect `ObjectStorageQueueFailedFiles`. The alternative is to be super organized and ensure anything that may affect it lives within the same test so that they're never parallelized.

Reply:
> There are other 4 tests running on the same ClickHouse DB that could potentially affect ObjectStorageQueueFailedFiles

But AFAIK no other tests suit apart from `test_storage_s3_queue` uses S3Queue (AzureQueue tests are in the same `test_storage_s3_queue`), so as you wrote "We know for sure all tests in test_storage_s3_queue never run in parallel." leads to a conclusion that `ObjectStorageQueueFailedFiles` cannot be changed anywhere apart from this test itself when it is executing.

Reply:
[Rolled back ](https://github.com/ClickHouse/ClickHouse/pull/67035/commits/15e0033016eb0e23a7e6f512d5096e50863e3187) the strict check because I think you're right.

I swear at some point I got assertion errors on that check when running in parallel, but maybe it was not the exactly same setup I have now which is closer to the CI. In any case, sorry for the noise. I'm running in my dev machine a batch of tests to make sure this still works ok, just in case.

Reply:
The good news is that I collected a list of assertions that failed for the test [here](https://github.com/ClickHouse/clickhouse-private/issues/13110#issuecomment-2250520981) and this assert is not among them. It's also not the one in the description of the issue.

The TL;DR is that increasing from 10s to 20s the max_age is what fixes most of the asserts :roll_eyes:

Reply:
I've run it with 100% successful rate after +1000 test executions :rocket: